### PR TITLE
composer@2.2 2.2.22 (new formula)

### PR DESCRIPTION
**Pull request by edspc**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

This PR adds a separate composer formula with LTS versions only.

### CODE REVIEW

**Review by SMillerDev — CHANGES_REQUESTED** [review]
This does not follow our naming convention. See https://docs.brew.sh/Versions.

File: Formula/c/composer-lts.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  url "https://getcomposer.org/download/2.2.22/composer.phar"
+  sha256 "7d3500cc8c9a74b47e14103de150ac95c25ca227b39ffc89cb3a8b495b5db1d2"
+  license "MIT"
+
+  livecheck do
+    url "https://getcomposer.org/download/"
+    regex(%r{href=.*?/v?(\d+(?:\.\d+)+)/composer\.phar}i)
+  end
+
+  depends_on "php"
```
Comment (SMillerDev) [inline Formula/c/composer-lts.rb:13]: According to the download page it doesn't work with php, only with disabled PHP versions (up to 7.1)

File: Formula/c/composer-lts.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  desc "Dependency Manager for PHP (LTS version)"
+  homepage "https://getcomposer.org/"
+  url "https://getcomposer.org/download/2.2.22/composer.phar"
+  sha256 "7d3500cc8c9a74b47e14103de150ac95c25ca227b39ffc89cb3a8b495b5db1d2"
+  license "MIT"
+
+  livecheck do
+    url "https://getcomposer.org/download/"
+    regex(%r{href=.*?/v?(\d+(?:\.\d+)+)/composer\.phar}i)
+  end
```
Comment (SMillerDev) [inline Formula/c/composer-lts.rb:11]: This will indicate the wrong version

Comment (edspc) [reply Formula/c/composer-lts.rb]: Fixed

File: Formula/c/composer-lts.rb (hunk 1 above)
Comment (edspc) [reply Formula/c/composer-lts.rb:13]: It means that the LTS version also supports outdated versions of PHP.  
Please check the [source](https://github.com/composer/composer/blob/2.2/composer.json#L25).

**Comment by edspc** [comment]
> This does not follow our naming convention. See https://docs.brew.sh/Versions.

I used the Jenkins formula as an example of LTS. Should I change it to `composer@2.2`?

File: Formula/c/composer-lts.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
+class ComposerLts < Formula
+  desc "Dependency Manager for PHP (LTS version)"
+  homepage "https://getcomposer.org/"
+  url "https://getcomposer.org/download/2.2.22/composer.phar"
+  sha256 "7d3500cc8c9a74b47e14103de150ac95c25ca227b39ffc89cb3a8b495b5db1d2"
+  license "MIT"
+
+  livecheck do
+    url "https://getcomposer.org/download/"
+    regex(%r{href=.*?/v?(2\.2\.\d+)+/composer\.phar}i)
```
Comment (SMillerDev) [inline Formula/c/composer-lts.rb:10]: This just pins it to 2.2, what if the LTS is updated to 2.5 or something?

**Comment by SMillerDev** [comment]
> I used the Jenkins formula as an example of LTS. Should I change it to composer@2.2?

Yes, since Jenkins-LTS is a separate project and release. Here it is "2.2 will be updated longer"

**Review by SMillerDev — COMMENTED** [review]
Some things I'd like to know before approving:
- What is the support policy for this version?
- When is this version EOL?

**Review by samford — COMMENTED** [review]
The `livecheck` block looks good to me 👍

**Comment by biu0w0** [comment]
> Some things I'd like to know before approving:
> 
> * What is the support policy for this version?
> * When is this version EOL?

Composer's 2.2 release is a LTS version.

Bugfixes for critical bugs and security issues until **at least end of 2023**, and will reassess then according to the usage.

REF: https://github.com/composer/composer/issues/10340#issue-1074236827

**Comment by SMillerDev** [comment]
> Bugfixes for critical bugs and security issues until at least end of 2023, and will reassess then according to the usage.

We could add the formula with a `disabled date: "2023-12-31"`. Although I'm not sure how useful it would be to have a formula for 2 months.

**Pull request closed without merging**